File: Java_Codes/10. Packages & Access Modifiers/mypackage/Employee2.java
package mypackage;

import java.time.LocalDate;

// Employee class with different access levels
public class Employee2 {
    public String name; // Public - Accessible everywhere
    private int age; // Private - Accessible only inside this class
    protected LocalDate joiningDate; // Protected - Accessible within the package and subclasses
    String department; // Default - Accessible within the package only

    // Constructor
    public Employee2(String name, int age, LocalDate joiningDate, String department) {
        this.name = name;
        this.age = age;
        this.joiningDate = joiningDate;
        this.department = department;
    }

    // Public method (accessible everywhere)
    public void showDetails() {
        System.out.println("Name: " + name);
        System.out.println("Age: " + age); // Allowed because it's inside the same class
        System.out.println("Joining Date: " + joiningDate);
        System.out.println("Department: " + department);
    }
}

File: Python_Codes/10. Packages & Access Modifiers/mypackage/Employee2.py
from datetime import date

class Employee2:
    def __init__(self, name: str, age: int, joining_date: date, department: str):
        self.name = name                  # Public
        self.__age = age                  # Private (name mangling)
        self._joining_date = joining_date # Protected (convention)
        self.department = department      # Default (treated as public in Python)

    # Public method
    def show_details(self):
        print("Name:", self.name)
        print("Age:", self.__age)  # Accessing private attribute within the class
        print("Joining Date:", self._joining_date)
        print("Department:", self.department)
